Modules/mathmodule.c

```diff
@@ -2471,14 +2474,60 @@ vector_norm(Py_ssize_t n, double *vec, double max, int found_nan)
             x = vec[i];
             assert(Py_IS_FINITE(x) && fabs(x) <= max);
             x *= scale;
-            x = x*x;
             assert(x <= 1.0);
+
+            t = x * T27;
+            hi = t - (t - x);
+            lo = x - hi;
+            assert (hi + lo == x);
+
+            x = hi * hi;
+            assert(csum >= x);
+            oldcsum = csum;
+            csum += x;
+            frac += (oldcsum - csum) + x;
+
+            x = 2.0 * hi * lo;
+            assert(csum >= x);
```

Comment:
After a split, `hi` has the same sign as the input, but `lo` may have a different sign.  For example,

```py
>>> split(1.0 - 2**-52)
(1.0, -2.220446049250313e-16)
````

In such cases , `2.0 * hi * lo` is negative, so the assert here isn't checking the _intended_ invariant.

Reply:
Right.  The intended invariant is |csum| >= |x|.   Since *csum* is always positive, I can omit the *fabs()* call.  I knew that *lo* is negative about half of the time, but forgot to add back the *fabs()*.